include/inviwo/core/util/observer.h

```diff
@@ -278,7 +278,9 @@ void Observable<T>::startBlockingNotifications() {
 }
 template <typename T>
 void Observable<T>::stopBlockingNotifications() {
-    --notificationsBlocked_;
+    if (notificationsBlocked_ != 0) {
```

Comment:
this should never happen if it does that is an error. there should not be an if here. In anything an assert.

Reply:
I called it by mistake instead of `startBlockingNotifications` and it "worked as expected" (i.e. the notifications stop coming). 
I agree we shouldn't call it and use the NoficationBlocker instead, I just did see the NoficationBlocker since I just glanced over the interface of observer interface and saw stop....

Reply:
so if you would have had that if there, then you would never have noticed your mistake...

Reply:
> 
> 
> so if you would have had that if there, then you would never have noticed your mistake...

Well, I noticed later when making the other commit that I had used the wrong function. If the if-statement would have been there my mistake would have been spotted earlier since the notifications wouldn't be blocked and my original issue would still have been there

Reply:
Anyway, what about making the start/Stop blocking private and only allow blocking using the `NoficationBlocker `?

Reply:
I would use an assert